Default lsp-prefer-flymake to t and fix enabling of flymake/flycheck

MaskRay opened the pull request:
See companion change https://github.com/emacs-lsp/lsp-ui/pull/202

yyoncho commented:
Looks good. Can you update the readme as well? It says that flyckeck is the default.

MaskRay commented:
Updated.

I changed `As you type reporting of parsing and compilation errors` to simplify `diagnostics` as `textDocument/publishDiagnostics` is not necessarily triggered by a `textDocument/didChange` on the server side.
